### PR TITLE
Compatibility for 2.0.3 Fody Plus Updated Dependencies, and VS2015/Win10 Compatibility

### PR DESCRIPTION
**Pull request by CDAGaming**
Self-Explanitory
Expanded from https://github.com/Fody/Costura/pull/187

Enjoy 👍

### CODE REVIEW

**Comment by shravan2x** [comment]
@CDAGaming Good work! I saw the `4.6.1` updates, but wasn't sure if that would significantly break compatibility.

**Comment by CDAGaming** [comment]
I don't think it should, as build is not complaining at all anymore, except for an architecture mismatch to x86, which also shouldn't be an issue

**Comment by CDAGaming** [comment]
I may also mention that i would be able to update the sdk but i don't think that'll be really neccesary for this project at this time

**Comment by CDAGaming** [comment]
PS Tip in advance as you'll see in this PR, Do NOT do ..\packages, and instead use $(SolutionDir)\packages to help reduce errors in the future from dependencies as @shravan2x saw from their previous test on the original pr

**Comment by shravan2x** [comment]
This PR definitely fixed almost all issues I had. There were some last missing usings in the Tests that I've fixed in https://github.com/shravan2x/Costura-v2-update/commit/c1ba4dbfabbf82c9d5b9e320cdd1e86822bfcd31.

Everything builds properly now!

**Comment by CDAGaming** [comment]
Ok, will accept the PR once I get home, thanks :)

**Comment by CDAGaming** [comment]
Seems this PR is READY TO MERGE at last, unless more testing requires other fixes not noticed as of yet :)

**Comment by SimonCropp** [comment]
@distantcam can u handle this one?

**Comment by distantcam** [comment]
@SimonCropp I'll try to get to this today.

**Comment by CDAGaming** [comment]
@distantcam DO NOT MERGE YET, have to add support now for 2.0.3, bc Fody decided to release (Just Now) a 2nd update

**Comment by CDAGaming** [comment]
This PR Will be adjusted to now support 2.0.3 versions

**Comment by SimonCropp** [comment]
the recent patch releases of fody have not changed the fody.cecil. so no need to update

**Comment by CDAGaming** [comment]
I dont know how, but it is erroring on 2.0.3 like crazy, but it all seems to point to:

`Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1503	Argument 1: cannot convert from 'string' to 'Mono.Cecil.AssemblyNameReference'	Fody	E:\Files\Costra-v2\Costura-v2-update\Fody\MsCoreReferenceFinder.cs	15	Active
`

Hope this is able to be resolved quickly so i can commit the 2.0.3 changes(Also Removed a few old references too)

File: Fody/Fody.csproj [hunk 1]
```diff
@@ -38,19 +38,19 @@
...
     </Reference>
     <Reference Include="Mono.Cecil.Mdb, Version=0.9.5.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\Lib\Cecil\Mono.Cecil.Mdb.dll</HintPath>
+      <HintPath>$(SolutionDir)\Lib\Cecil\Mono.Cecil.Mdb.dll</HintPath>
     </Reference>
     <Reference Include="Mono.Cecil.Pdb, Version=0.9.5.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\Lib\Cecil\Mono.Cecil.Pdb.dll</HintPath>
+      <HintPath>$(SolutionDir)\Lib\Cecil\Mono.Cecil.Pdb.dll</HintPath>
```
Comment (SimonCropp) [inline Fody/Fody.csproj]: this should reference the fody.cecil nuget package

Comment (CDAGaming) [inline Fody/Fody.csproj]: @SimonCropp Yea, i did that with the 2.0.3 Commit, which ill be pushing here now, but i do not know why these errors are coming up now with 2.0.3 but not on 2.0.2

**Comment by CDAGaming** [comment]
Build Error Log, seems to be on the same file

**Comment by SimonCropp** [comment]
variable defs no longer have name param. just remove it

**Comment by CDAGaming** [comment]
As I'm not on my PC ATM, can you pr me the necessary changes so it'll be easier to merge together tomorrow and

**Comment by CDAGaming** [comment]
Commits in a bit

**Comment by CDAGaming** [comment]
Wow, Microsoft wants me to do a ton of simplifications on this(Maybe another time for another PR)

**Comment by CDAGaming** [comment]
Ok, so quite a bit of code concerning Sequence point and MSCoreReferences were commented out in order for the build to succeed, so you may need to test this as well, commiting in a moment

**Comment by distantcam** [comment]
The MSCoreReferences are needed. Currently the tests are not passing.

**Comment by distantcam** [comment]
Manually merged and released in 1.4.0

**Comment by CDAGaming** [comment]
Thanks, had to go to sleep, so I had limited time

**Comment by CDAGaming** [comment]
Hm, noticing on your commits you added app.config, but I don't see that file as entirely needed

**Comment by CDAGaming** [comment]
It also removed the .. to $(SolutionDir) change that gave me lots of issues with both me and shraven with finding the Dependency files in vs before fixing that with $(SolutionDir)/packages

**Comment by distantcam** [comment]
Yeah, that should be managed by NuGet. The error shraven posted was that VS2017 upgraded the C++ projects to .NET 4.6.1, which caused all the 4.5 projects referencing them to complain.

**Comment by CDAGaming** [comment]
Ah, Okay, and how about app.config, which can usually be added by mistake when a NuGet package is changed, but is not really necessary to have unless you have your own changes in that file that requires it

**Comment by distantcam** [comment]
Yeah, I'm gonna delete those in a commit soon.

**Comment by CDAGaming** [comment]
Ok, thanks again, happy to help work on this too, maybe some time I'll commit more things(Maybe those MS simplifications as there were a ton of them, heh)

**Pull request closed without merging**